### ISSUE TITLE
slow reaction when typing in fullscreen mode

### ISSUE THREAD

**Issue by J-Saeedi** [xournalpp/xournalpp#3941]
when I try to insert text, if xournalpp's window was small, there is no problem. but as I switch to full screen mode, inserting text became with lag.

```
$xournalpp --version
xournalpp 1.1.1+dev
└──libgtk: 3.24.30
```
OS:  kubuntu 21.10.
I donno how I can show this lag with screenshot or screen records. Tell me everything you need to see.
Also there is no problem with xournal (old program) [http://xournal.sourceforge.net/]

**Comment by rolandlo** [comment]
That issue will probably be fixed when #3942 is merged.

**Comment by Technius** [comment]
@J-Saeedi Do you still encounter this issue on the latest nightly?

**Comment by J-Saeedi** [comment]
@Technius Yes. I also tested the new version and there is still a problem. when I try to write a text in fullscreen mode, it became laggy. (slow reaction to keyboard inputs.) I also tried changing Input Devices from preferences and results to nothing better.

This is the log:
```┌─[w-admin@parrot]─[~/Desktop/test-xournalpp]
└──╼ $./xournalpp-1.1.1+dev-nightly.20220416-x86_64.AppImage 
/usr/lib/x86_64-linux-gnu/gvfs/libgvfscommon.so: undefined symbol: g_task_set_name
Failed to load module: /usr/lib/x86_64-linux-gnu/gio/modules/libgvfsdbus.so

(AppRun.wrapped:21143): Gtk-WARNING **: 09:37:20.729: Theme parsing error: gtk.css:2:123: Failed to import: Error opening file /home/w-admin/.local/share/gnome-shell/extensions/unite@hardpixel.eu/styles/gtk3/buttons-left/maximized.css: No such file or directory
Gtk-Message: 09:37:20.779: Failed to load module "colorreload-gtk-module"
Gtk-Message: 09:37:20.780: Failed to load module "window-decorations-gtk-module"
Gtk-Message: 09:37:20.781: Failed to load module "appmenu-gtk-module"
** Message: 09:37:20.784: TEXTDOMAINDIR = /tmp/.mount_xourna8NWhiU/usr/share/locale:, Platform-specific locale dir = /usr/share/xournalpp/../locale, chosen directory = /tmp/.mount_xourna8NWhiU/usr/share/locale
ALSA lib pcm.c:2660:(snd_pcm_open_noupdate) Unknown PCM cards.pcm.rear
ALSA lib pcm.c:2660:(snd_pcm_open_noupdate) Unknown PCM cards.pcm.center_lfe
ALSA lib pcm.c:2660:(snd_pcm_open_noupdate) Unknown PCM cards.pcm.side
ALSA lib pcm_route.c:877:(find_matching_chmap) Found no matching channel map
ALSA lib pcm_route.c:877:(find_matching_chmap) Found no matching channel map
ALSA lib pcm_route.c:877:(find_matching_chmap) Found no matching channel map
ALSA lib pcm_route.c:877:(find_matching_chmap) Found no matching channel map
Cannot connect to server socket err = No such file or directory
Cannot connect to server request channel
jack server is not running or cannot be started
JackShmReadWritePtr::~JackShmReadWritePtr - Init not done for -1, skipping unlock
JackShmReadWritePtr::~JackShmReadWritePtr - Init not done for -1, skipping unlock
Cannot connect to server socket err = No such file or directory
Cannot connect to server request channel
jack server is not running or cannot be started
JackShmReadWritePtr::~JackShmReadWritePtr - Init not done for -1, skipping unlock
JackShmReadWritePtr::~JackShmReadWritePtr - Init not done for -1, skipping unlock
ALSA lib pcm_oss.c:377:(_snd_pcm_oss_open) Unknown field port
ALSA lib pcm_oss.c:377:(_snd_pcm_oss_open) Unknown field port
ALSA lib pcm_usb_stream.c:486:(_snd_pcm_usb_stream_open) Invalid type for card
ALSA lib pcm_usb_stream.c:486:(_snd_pcm_usb_stream_open) Invalid type for card
Cannot connect to server socket err = No such file or directory
Cannot connect to server request channel
jack server is not running or cannot be started
JackShmReadWritePtr::~JackShmReadWritePtr - Init not done for -1, skipping unlock
JackShmReadWritePtr::~JackShmReadWritePtr - Init not done for -1, skipping unlock
** Message: 09:37:20.966: Plugin "MigrateFontSizes" UI initialized

** (AppRun.wrapped:21143): WARNING **: 09:37:21.193: Cannot update zoomPresentationValue yet. This should only happen on startup! 
** Message: 09:40:21.621: Info: autosave document...
** Message: 09:40:21.621: Autosaving to /home/w-admin/.cache/xournalpp/autosaves/21143.autosave.xopp
^C
```

**Comment by Technius** [comment]
Are you using nvidia drivers? If so, this may be related to #1744.

**Comment by J-Saeedi** [comment]
@Technius Yes, I am using Nvidia drivers. Now, what should I do? (I'm new to GitHub's issues)

**Comment by Technius** [comment]
Could you try using Intel drivers only?

**Comment by tattsan** [comment]
A slight delay is observed in a non-Nvidia environment. When the zoom magnification is set to maximum and a wide letter like `W` is typed in succession, the right half of the last letter may not be displayed for about 0.5 seconds. It does not seem to be so bad as in #1744. This is not affected by window size, but by zoom magnification.

**Comment by J-Saeedi** [comment]
@Technius At the moment it is not possible for me to change my system drivers.
Can I do this by adding a bash parameter?

**Comment by tattsan** [comment]
I'll post a screenshot. This happens the same way regardless of whether you are in an Nvidia environment or not.

https://user-images.githubusercontent.com/6355486/164603514-8c0ada12-3604-443d-945b-b809f5ef19b2.mp4

@J-Saeedi , is this the same behavior as your issue?

P.S.
I have replaced only the Pen icon with the old one, because I sometimes mistake it for the highlighter if it is the same color as the highlighter.

**Comment by J-Saeedi** [comment]
@tattsan Yes, it was as same as yours.
this video may show lags better: (I hold the `w` key and record, first in Xournal++ and then in a text editor [kate])

https://user-images.githubusercontent.com/103021017/164611141-22360bd2-d9a2-4c9e-9f71-05b32caae365.mp4

**Comment by tattsan** [comment]
It seems that it is not the same issue. In my video, I typed letters one by one. If I input these letters successively via key repeat, there are  no lags between letters. My issue is that half of the characters are initially displayed missing and there is a lag before all are displayed.

**Comment by bhennion** [comment]
This may be improved by #4074.

An additional question: do you have a high DPI monitor? and a HiDPI scaling?
(to answer the scaling question, you could get the information by running `echo $GDK_SCALE` in a terminal)

**Comment by J-Saeedi** [comment]
At first, I clone the repo, then I switched to  `release-1.1` and I followed the guide for compiling the code.
after installing the .DEB package, I tested the fresh version like the way in the [previous comment](https://github.com/xournalpp/xournalpp/issues/3941#issuecomment-1106034231):
> this video may show lags better: (I hold the `w` key and record, first in Xournal++ and then in a text editor [kate])
>  Kazam_screencast_00009.mp4

but it is still laggy. another point that I noticed is when I am typing in xournal++, CPU and GPU usage increase.
for HiDPI, because I am using KDE Plasma Desktop, GDK_SCALE returns nothing, but I guess these two screenshots may be helpful:
![image](https://user-images.githubusercontent.com/103021017/173208780-3ad548d6-a533-4142-91d4-32cb6f6b2332.png)
![image](https://user-images.githubusercontent.com/103021017/173208897-58715d2d-face-4d56-beab-5c14e4c90337.png)
